src/racer.c

```diff
@@ -975,42 +1029,138 @@ void func_800570A4(Object *obj, s32 arg1, s32 arg2) {
-    s32 phi_t0;
-    Object_Racer *obj64;
+    s32 temp_v1;
+    Object_64 *temp_s2;
 
-    obj64 = &obj->unk64->racer;
-    if ((obj64->unk108 == 0) && ((!(arg3 & 0x80)) || (D_8011D55C != -1))) {
+    temp_s2 = obj->unk64;
+    if ((temp_s2->racer.unk108 == 0) && ((!(arg3 & 0x80)) || (D_8011D55C != -1))) {
         if (arg3 == 2) {
-            if ((obj64->unk24 != 0) && (arg1 != obj64->unk2A)) {
-                func_800096F8(obj64->unk24);
-                obj64->unk24 = 0;
+            if ((temp_s2->racer.unk24 != 0) && (arg1 != temp_s2->racer.unk2A)) {
+                func_800096F8(temp_s2->racer.unk24);
+                temp_s2->racer.unk24 = 0;
             }
         }
-        if (obj64->unk24 == 0) {
-            if (arg3 != 3 || get_random_number_from_range(0, 1) != 0) {
-                obj64->unk2A = arg1;
-                arg1 += obj64->characterId;
-                arg2--;
-                phi_t0 = (get_random_number_from_range(0, arg2) * 12) + arg1;
-                if (arg2 > 0) {
-                    while (phi_t0 == obj64->unk28) {
-                        phi_t0 = (get_random_number_from_range(0, arg2) * 12) + arg1;
-                    }
+        if ((temp_s2->racer.unk24 == 0) && ((arg3 != 3) || (get_random_number_from_range(0, 1) != 0))) {
+            temp_s2->racer.unk2A = arg1;
+            arg1 += temp_s2->racer.characterId;
+            temp_v1 = (get_random_number_from_range(0, arg2 - 1) * 12) + arg1;
+            if (arg2 - 1 > 0) {
+                while (temp_v1 == temp_s2->racer.unk28) {
+                    temp_v1 = (get_random_number_from_range(0, arg2 - 1) * 12) + arg1;
                 }
-                //sp34 = phi_t0;
-                func_80009558(phi_t0, obj->segment.trans.x_position, obj->segment.trans.y_position, obj->segment.trans.z_position, 4, &obj64->unk24);
-                obj64->unk28 = phi_t0;
             }
+            func_80009558(temp_v1, obj->segment.trans.x_position, obj->segment.trans.y_position, obj->segment.trans.z_position, 4, &temp_s2->racer.unk24);
+            temp_s2->racer.unk28 = temp_v1;
         }
     }
 }
-#else
-GLOBAL_ASM("asm/non_matchings/racer/func_800570B8.s")
-#endif
 
-GLOBAL_ASM("asm/non_matchings/racer/func_80057220.s")
+f32 func_80057220(Object* arg0, Object_Racer* arg1) {
+    f32 speedMultiplier;
+    s32 temp_v1;
+    f32 bananas;
+    f32 cap;
+    s32 phi_a1;
+    s32 sp28;
+
+    speedMultiplier = 1.0f;
+    if (D_8011D540 != 0) {
+        speedMultiplier = 0.0f;
+    }
+    sp28 = func_800113AC();
+    if (D_8011D540);
+    if ((D_8011D540 > 0) && (D_8011D540 < 30) && (!arg1->unk1F4)) {
+        temp_v1 = D_8011D540 - 14;
+        if ((gActivePlayerButtonPress & 0x8000) != 0) {
+            if ((temp_v1 < 0) && (sp28 >= 0)) {
+                temp_v1 = 0;
+            }
+            if (temp_v1 < 0) {
+                temp_v1 = -temp_v1;
+            }
+
+            if ((gCurrentCarInput & 0x2000)) {
```

Comment:
```suggestion
            if ((gCurrentCarInput & Z_TRIG)) {
```

Reply:
I'm assuming gCurrentCarInput is a button press of course. We could  even make a define just renaming Z_TRIG to horn as well if we feel like it.